### DISCUSSION TITLE
Read-only: Read-only data container

### DISCUSSION THREAD

**Discussion by Refat1979** [dbeaver/dbeaver#35543] [Q&A] [answered]
### Description

![IMG-20240814-WA0003](https://github.com/user-attachments/assets/293b0938-0edd-4b55-ac7a-473fc446c6ed)

### DBeaver Version

24.1.4

### Operating System

macOS Sonoma 14.5

### Database and driver

Trying to edit columns and run queries but failed due to the read only 

### Additional context

Tried to check the connection and it was all good

**Comment by Refat1979** [comment]
Same thing on windows

**Comment by ask9** [comment]
Turn off Read-only mode on the main panel for your connection:

https://github.com/user-attachments/assets/d0128bd5-470e-42f5-9113-76316669c295

**Comment by Refat1979** [comment]
It's disabled but still the same photo attached
![IMG-20240814-WA0013](https://github.com/user-attachments/assets/e2447501-fefe-47fc-97a8-14db914b49d3)
, thanks for responding

**Comment by ask9** [comment, marked as answer]
CSV-driver does not allow editing data.

> CsvJdbc accepts only SQL SELECT queries from a single table and does not support INSERT, UPDATE, DELETE or CREATE statements.

https://csvjdbc.sourceforge.net/doc.html

You can find it out, if you click "Driver documentation" link in "Connection configuration" window on the tab "Driver properties".

![изображение](https://github.com/user-attachments/assets/80a21904-047f-417b-9f4e-8424a2770803)
